kafka-connect-s3/src/main/java/io/confluent/connect/s3/auth/AWSAssumeRoleCredentialsProvider.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Nordstrom, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.confluent.connect.s3.auth;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import org.apache.kafka.common.Configurable;
+
+import java.util.Map;
+
+public class AWSAssumeRoleCredentialsProvider implements AWSCredentialsProvider, Configurable {
+
+  public static final String EXTERNAL_ID_CONFIG = "external.id";
+  public static final String ROLE_ARN_CONFIG = "role.arn";
+  public static final String SESSION_NAME_CONFIG = "session.name";
```

Comment:
would role session name be a better name for this property (maps better with the [CLI](https://docs.aws.amazon.com/cli/latest/reference/sts/assume-role.html)):

```suggestion
  public static final String SESSION_NAME_CONFIG = "role.session.name";
```

Reply:
Seemed like a good idea to prefix all parameters with 'role'.

Reply:
Prefixing `external.id` does not have the same symmetry with the CLI, unless the plan is changed to take the literal `assume.role` prefix.

```
{
    "s3.credentials.provider.class": "io.confluent.connect.s3.auth.AWSAssumeRoleCredentialsProvider",
    "s3.credentials.provider.assume.role.arn": "arn:aws:iam::012345678901:role/my-restricted-role",
    "s3.credentials.provider.assume.role.session.name": "my-session-name",
    "s3.credentials.provider.assume.role.external.id": "my-external-id"
}